TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteListFragment.kt

```diff
@@ -78,10 +79,13 @@ class NoteListFragment : Fragment(), AdapterView.OnItemClickListener, AdapterVie
 
         view.findViewById<FloatingActionButton>(R.id.add_fab).setOnClickListener {
             // Set selected item to newly created note (first element in list)
-            startNoteFragment(addInode(ROOT))
+            val inode = fileHandler.addInode(ROOT)
+            arrayAdapter?.notifyDataSetChanged()
+            startNoteFragment(inode)
         }
 
-        loadDirectory(ROOT)
+        fileHandler.loadDirectory(requireContext(), ROOT)
+        arrayAdapter?.notifyDataSetChanged()
```

Comment:
and i'm not sure this has to be called unless you want the UI to be updated immediately, otherwise think whenever something else in the UI changes the adapter might already be notified? please disregard if that's not what you saw happening haha